### PR TITLE
[nnx] improve toy examples

### PR DESCRIPTION
**Pull request by cgarciae**
# What does this PR do?

Improves toy examples, makes them more idiomatic according to current standards.

### CODE REVIEW

File: flax/experimental/nnx/examples/toy_examples/01_functional_api.py [hunk 1]
```diff
@@ -67,10 +67,11 @@ def train_step(params, counts, batch):
   x, y = batch
 
   def loss_fn(params):
-    y_pred, (_, updates) = static.apply(params, counts)(x)
-    counts_ = updates.extract(Count)
+    model = static.merge(params, counts)
+    y_pred = model(x)
+    _counts = model.extract(Count)
```
Comment (superbobry) [inline flax/experimental/nnx/examples/toy_examples/01_functional_api.py:72]: Why the `_` prefix? It is normally used (by convention) to indicate that a variable is unused, but here this isn't the case.

Comment (cgarciae) [reply flax/experimental/nnx/examples/toy_examples/01_functional_api.py:72]: Should we keep `counts_` then?  Here we need a new variable name, cannot reuse `counts`.

Comment (superbobry) [reply flax/experimental/nnx/examples/toy_examples/01_functional_api.py:72]: extracted_counts or new_counts?

Comment (cgarciae) [reply flax/experimental/nnx/examples/toy_examples/01_functional_api.py:72]: `new_counts` it is

**Review by superbobry — APPROVED** [review]

**Pull request merged**